include/mbgl/util/run_loop.hpp

```diff
@@ -64,14 +65,23 @@ class RunLoop : private util::noncopyable {
     template <class Fn, class Cb, class... Args>
     std::unique_ptr<WorkRequest>
     invokeWithCallback(Fn&& fn, Cb&& callback, Args&&... args) {
-        auto flag = std::make_shared<bool>();
+        auto flag = std::make_shared<std::atomic<bool>>();
```

Comment:
I would but I don't think it's possible to portably use `std::shared_ptr<std::atomic_flag>` due to the initialization requirements of `ATOMIC_FLAG_INIT`.